cpp/src/arrow/compute/kernels/filter.h

```diff
@@ -50,6 +50,42 @@ ARROW_EXPORT
 Status Filter(FunctionContext* ctx, const Array& values, const Array& filter,
               std::shared_ptr<Array>* out);
 
+/// \brief Filter a chunked array with a boolean selection filter
+///
+/// The output chunked array will be populated with values from the input at positions
+/// where the selection filter is not 0. Nulls in the filter will result in nulls
+/// in the output.
+///
+/// For example given values = ["a", "b", "c", null, "e", "f"] and
+/// filter = [0, 1, 1, 0, null, 1], the output will be
+/// = ["b", "c", null, "f"]
+///
+/// \param[in] ctx the FunctionContext
+/// \param[in] values chunked array to filter
+/// \param[in] filter indicates which values should be filtered out
+/// \param[out] out resulting chunked array
+ARROW_EXPORT
+Status Filter(FunctionContext* ctx, const ChunkedArray& values, const Array& filter,
+              std::shared_ptr<ChunkedArray>* out);
```

Comment:
Per comments on Take below I think that using Datum and having fewer public APIs would be better. There are implicit ctors for Datum to make the usage easier

Reply:
I briefly looked into refactoring to use Datum but it didn't seem like a good use of my time right now to figure it out. https://issues.apache.org/jira/browse/ARROW-7009 is for someone else to pick that up.

Reply:
No problem